File: 101_200/145_Binary_Tree_Postorder_Traversal/Solution.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> res;
        stack<TreeNode*> call;
        if (root != nullptr) call.push(root);
        while(!call.empty()){
            TreeNode *t = call.top();
            call.pop();
            if(t != nullptr){
                call.push(t);  
                call.push(nullptr); 
                if(t->right) call.push(t->right);
                if(t->left) call.push(t->left);
            }else{
                res.push_back(call.top()->val);
                call.pop();
            }
        }
        return res;   
    }
};

File: 101_200/145_Binary_Tree_Postorder_Traversal/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        ans, st = [], []
        if not root: return ans
        st.append(root)
        while len(st) > 0:
            t = st.pop()
            if t:
                st.append(t)
                st.append(None)
                if t.right: st.append(t.right)
                if t.left: st.append(t.left)
            else:
                ans.append(st.pop().val)
        return ans
